src/commands/server/restore.ts

```diff
@@ -83,12 +113,204 @@ export default class Restore extends Command {
+          }
+
+          flags.version = backupCr.status.cheVersion
+          flags['snapshot-id'] = backupCr.status.snapshotId
+          task.title = `${task.title}OK`
+        },
+      },
+      {
+        title: 'Getting backup server info...',
+        task: async (ctx: any, task: any) => {
+          // Get all information about backup server and validate it where possible
+          // before redeploying operator to requested version.
+          ctx.backupServerConfig = getBackupServerConfiguration(flags)
+          task.title = `${task.title}OK`
+        },
+      },
+      {
+        title: 'Getting installation templates...',
+        enabled: (ctx: any) => flags.version && ctx.currentOperatorVersion !== flags.version && !flags.templates,
+        task: async (ctx: any, task: any) => {
+          if (flags.installer === 'olm') {
+            // Using embedded templates here as they are used in install flow for OLM.
+            // OLM install flow should be reworked to not to use templates, but OLM dependencies.
+            flags.templates = getEmbeddedTemplatesDirectory()
+          } else {
+            const getTemplatesTasks = new Listr(undefined, ctx.listrOptions)
+            getTemplatesTasks.add(checkChectlAndCheVersionCompatibility(flags))
+            getTemplatesTasks.add(downloadTemplates(flags))
+            return getTemplatesTasks
+          }
+          task.title = `${task.title}OK`
+        },
+      },
+      {
+        title: 'Remove current Che operator',
+        enabled: (ctx: any) => ctx.isOperatorDeployed && flags.installer === 'olm' && flags.version && ctx.currentOperatorVersion !== flags.version,
+        task: async (ctx: any, _task: any) => {
+          // All preparations and validations must be done before this task!
+          // Delete old operator if any in case of OLM installer.
+          // For Operator installer, the operator deployment will be downgraded if needed.
+          const olmTasks = new OLMTasks()
+          const olmDeleteTasks = olmTasks.deleteTasks(flags)
+          return new Listr(olmDeleteTasks, ctx.listrOptions)
+        },
+      },
+      {
+        title: 'Deploy requested version of Che operator',
+        enabled: (ctx: any) => flags.version && ctx.currentOperatorVersion !== flags.version,
+        task: async (ctx: any, _task: any) => {
+          const isOpenshift = await kube.isOpenShift()
+          // Set defaults for some parameters if they weren't set
+          if (isOpenshift) {
+            flags.platform = 'openshift'
+            flags['cluster-monitoring'] = true
+          } else {
+            flags.platform = 'kubernetes'
+            // TODO domain ?
+          }
+
+          // Use plain operator on Kubernetes or if it is requested instead of OLM
+          if (!isOpenshift || flags.installer === 'operator') {
+            const operatorTasks = new OperatorTasks()
+            // Update tasks can deploy operator as well
```

Comment:
// We use update tasks to restore corresponding objects (which are not backed up)